db/config.cc

```diff
@@ -472,6 +472,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , commitlog_total_space_in_mb(this, "commitlog_total_space_in_mb", value_status::Used, -1,
         "Total space used for commitlogs. If the used space goes above this value, Scylla rounds up to the next nearest segment multiple and flushes memtables to disk for the oldest commitlog segments, removing those log segments. This reduces the amount of data to replay on startup, and prevents infrequently-updated tables from indefinitely keeping commitlog segments. A small total commitlog space tends to cause more flush activity on less-active tables.\n"
         "Related information: Configuring memtable throughput")
+    , commitlog_compression(this, "commitlog_compression", value_status::Used, {}, 
+        "Compression to apply to the commit log. If omitted, the commit log will be written uncompressed.  LZ4, Snappy, Zstd and Deflate compressors are supported")
```

Comment:
Please make it LiveUpdate.

There should be separate options for the compressor name and for enable/disable, since we want to be able to disable compression but remember the compressor for decompressing during replay.

Reply:
Compressor is stored per entry, so no need for a separate enable/disable. Setting commitlog_compression="" will not affect the readability of already created entries. 
Live update is somewhat tricky since the compressor object is held inside the replica::database object, but can add listeners I guess...

Reply:
Sorry, meant commitlog_compression={}

Reply:
Maybe one of the changelogs is misleading on that. Yes, I saw later you encode the compression used.

If it's written once per segment, we must make sure the config is only sampled once per segment.

Reply:
The thing is that compressor parameters are only useful for compressing. Decompression can always resolve and operate assuming the correct compressor, so we only need the tag value to determine which compressor, if any, an entry uses.